### ISSUE TITLE
Missing service account after 0.1.57 compliance operator upgrade

### ISSUE THREAD

**Issue by bukovjanmic** [ComplianceAsCode/compliance-operator#177]
After recent compliance operator 0.1.57 upgrade, rerunner job are failing with the following message:

Error creating: pods "cis-compliance-rerunner-27797820-" is forbidden: error looking up service account openshift-compliance/rerunner: serviceaccount "rerunner" not found

It seems the operator failed to create necessary service account for the CronJob cis-compliance-rerunner, which wants to run under this service account, or the template should use a different SA.

**Comment by bukovjanmic** [comment]
It seems the SA is created on new operator installation, but somehow is removed when there is upgrade from previous version.

**Comment by jhrozek** [comment]
Sorry for the late reply. I'm going to try to reproduce this. Two questions:

- are you using the upstream catalogs and builds or through OCP (aka RH builds)
- if you still have a cluster around that's broken, could you paste the results of `oc describe compliance-operator.v0.1.57`. There might still be a pod corresponding to the upgrade in the `openshift-marketplace` NS, is there anything interesting there?

btw one thing that I think might be related is that the offending SA is not owned by the CSV, but others are:
```
- apiVersion: v1
  imagePullSecrets:
  - name: remediation-aggregator-dockercfg-57cdw
  kind: ServiceAccount
  metadata:
    creationTimestamp: "2022-11-09T10:53:11Z"
    labels:
      operators.coreos.com/compliance-operator.openshift-compliance: ""
    name: remediation-aggregator
    namespace: openshift-compliance
    ownerReferences:
    - apiVersion: operators.coreos.com/v1alpha1
      blockOwnerDeletion: false
      controller: false
      kind: ClusterServiceVersion
      name: compliance-operator.v0.1.57
      uid: b697d6bc-0ddd-44cd-8324-30a9e6db9951
    resourceVersion: "33087"
    uid: 36ecc10f-b44f-442b-a017-3d09c56ca6f9
  secrets:
  - name: remediation-aggregator-dockercfg-57cdw
- apiVersion: v1
  imagePullSecrets:
  - name: rerunner-dockercfg-d5jsq
  kind: ServiceAccount
  metadata:
    creationTimestamp: "2022-11-09T10:53:09Z"
    name: rerunner
    namespace: openshift-compliance
    resourceVersion: "32953"
    uid: 85512a49-0504-4d40-8024-9cdb591f934d
  secrets:
  - name: rerunner-dockercfg-d5jsq
```

**Comment by jhrozek** [comment]
going to dig into this more..

**Comment by jhrozek** [comment]
PR: https://github.com/ComplianceAsCode/compliance-operator/pull/179

**Comment by rhmdnd** [comment]
This was fixed in #179. Should be available in 0.1.58.
